### ISSUE TITLE
Mail with news files note leaving

### ISSUE THREAD

**Issue by FrRene06** [Cloud9Developer/Jellyfin-Newsletter-Plugin#112]
good morning,

first of all, thank you for the plug-in.

But although the mail test works, the mail with the news content doesn't leave.

I enclose the program log.

Kind regards,

Laurent
[2024-06-02_Newsletter.log](https://github.com/user-attachments/files/15524405/2024-06-02_Newsletter.log)

**Comment by Cloud9Developer** [comment]
Looks like it scanned, but had an error processing every item? Can you turn on debug mode, try again, and send me new logs?

**Comment by FrRene06** [comment]
hello, Ok.
Thanx for your Help !

Laurent.
[Uploading 2024-06-03_Newsletter.log…]()

**Comment by mastamjay** [comment]
It's exactly the same for me after updating to Jellyfin 10.9.3 and NLP 0.6.4.2.

I had already posted in the [Jellyfin Plugin Forum](https://forum.jellyfin.org/t-jellyfin-newsletter-plugin?pid=24951#pid24951).

**Comment by p37307** [comment]
Not sure if this is what is going on with you, but I looked in the config file `jellyfin\plugins\configurations\Jellyfin.Plugin.Newsletters.xml` for Windows 10 and had to change `<NewsletterDir>` path from `\cache\temp/Newsletters/` to `\cache\temp\Newsletters\` and manually created the Newsletters folder.

It worked after that.

**Comment by mastamjay** [comment]
Thank you for your help. I have changed it, but unfortunately no newsletter is sent out. 
Test email works 100%.


Does the newsletter work with Jellyfin Server for Windows?

**Comment by Cloud9Developer** [comment]
> Not sure if this is what is going on with you, but I looked in the config file `jellyfin\plugins\configurations\Jellyfin.Plugin.Newsletters.xml` for Windows 10 and had to change `<NewsletterDir>` path from `\cache\temp/Newsletters/` to `\cache\temp\Newsletters\` and manually created the Newsletters folder.
> 
> It worked after that.

Ahh, looks like unix paths and windows don't mesh.. Can you open an issue for this? I'll see if I can find a way to get that fixed

**Comment by Cloud9Developer** [comment]
> Thank you for your help. I have changed it, but unfortunately no newsletter is sent out. Test email works 100%.
> 
> Does the newsletter work with Jellyfin Server for Windows?

@mastamjay Newsletter should work as long as jellyfin does, yeah. I looked at your logs but don't see anything helpful. Can you turn on debug mode in the plugin settings and try again, then share those logs?

**Comment by mastamjay** [comment]
Switch on debug mode
here is the newsletter log file after filesystem scraping
[Newsletter Log file.txt](https://github.com/user-attachments/files/15552203/Newsletter.Log.file.txt)

**Comment by Cloud9Developer** [comment]
@mastamjay , is that the whole logfile? It only shows an 1 error and then nothing after. It should continue past that 1 error (error processing file) and continue scanning

**Comment by mastamjay** [comment]
> @mastamjay , is that the whole logfile? It only shows an 1 error and then nothing after. It should continue past that 1 error (error processing file) and continue scanning
sry. I had only copied a part from the log. Here is the whole
[Full Newsletter Log.txt](https://github.com/user-attachments/files/15552661/Full.Newsletter.Log.txt)

**Comment by mastamjay** [comment]
FYI: With Jellyfin 10.8.x and the newsletter plugin v.0.6.0. everything worked fine so far.
I think that with version 0.6.4.2, the new media is not recognised and not made available for the email.

**Comment by nylimited** [comment]
Somehow I suspect there is more going on with the JF 10.0.x changes.  I run this plugin and Ombi side by side, generating 2 newsletters. Ombi has not been updated toward the new JF.  The Ombi newsletter included a few very old media and the plugin newsletter dropped a new one or two. I don't know why but there is a lot more debugging needed than meets the eye.  🫢

**Comment by Cloud9Developer** [comment]
What's throwing me for a loop is that it's only scanning one item then stopping. @mastamjay , are you clicking scan multiple times (starting it then stopping it)?

**Comment by mastamjay** [comment]
> What's throwing me for a loop is that it's only scanning one item then stopping. @mastamjay , are you clicking scan multiple times (starting it then stopping it)?
No, I just click Filesystem Scraper once under Scheduled Tasks. I then wait for the process to complete, which takes a maximum of 5 seconds. Then I wait another 10 seconds and click on Email Newsletter.

**Comment by mastamjay** [comment]
Should I completely uninstall the newsletter plugin from jellyfin and also some files/folders of the plugin under Windows?
If so, which folders and files of the plugin should I delete under Windows? Then I could test another version.
Or is it enough to choose another version or plugin/catalogue and install it?

**Comment by Cloud9Developer** [comment]
@mastamjay , can you try to do a full rescan and see what happens there? make a backup of your newsletters.db file in your /config directory, then delete the existing newsletters.db file. Then run the scan and email tasks

**Comment by nylimited** [comment]
Should the scraper run only once before the newsletter is created?  I kind of assumed (yes, I know about assume ..) that I can run it daily, accumulate new items and when the newsletter runs it sets everything "read" so to speak.  At the moment the newsletter runs only once a week (on Fridays) but the file system scraper runs daily. Since I have not pushed any new media to the system yet I can easily make adjustments.

**Comment by mastamjay** [comment]
> @mastamjay , can you try to do a full rescan and see what happens there? make a backup of your newsletters.db file in your /config directory, then delete the existing newsletters.db file. Then run the scan and email tasks

[New full Newsletter log.txt](https://github.com/user-attachments/files/15561756/New.full.Newsletter.log.txt)

**Comment by mastamjay** [comment]
I have noticed that the Newsletter.db file is only 28kb in size. The 0.6.0 file is over 1MB and contains the scraped data. I looked at it using Windows Notepad.

**Comment by Cloud9Developer** [comment]
Think I have an idea on what's happening. It's crashing when trying to get the BaseItems (jellyfin items). 

What's your file structure look like?

**Comment by mastamjay** [comment]
> Think I have an idea on what's happening. It's crashing when trying to get the BaseItems (jellyfin items).
> 
> What's your file structure look like?
What exactly do you mean by file structure?

**Comment by Cloud9Developer** [comment]
```Shows
├── Series (2010)
│   ├── Season 00
│   │   ├── Some Special.mkv
│   │   ├── Episode S00E01.mkv
│   │   └── Episode S00E02.mkv
│   ├── Season 01
│   │   ├── Episode S01E01-E02.mkv
│   │   ├── Episode S01E03.mkv
│   │   └── Episode S01E04.mkv
│   └── Season 02
│       ├── Episode S02E01.mkv
│       ├── Episode S02E02.mkv
│       ├── Episode S02E03 Part 1.mkv
│       └── Episode S02E03 Part 2.mkv
```

are you following this format with your media structure on your filesystem?

**Comment by mastamjay** [comment]
Yes, just with the name of the series before that and in German:
Example:
English wording
Series -> 3 Body Problem -> Season 1 | 3 Body Problem - S01E01- Countdown.mkv
German wording
Serien -> 3 Body Problem -> Staffel 1 | 3 Body Problem - S01E01- Countdown.mkv

**Comment by Cloud9Developer** [comment]
so it's `/series/3 Body Problem/Season 1/3 Body Problem - S01E01- Countdown.mkv`
Correct?

**Comment by mastamjay** [comment]
> so it's `/series/3 Body Problem/Season 1/3 Body Problem - S01E01- Countdown.mkv` Correct?
Yes

**Comment by Cloud9Developer** [comment]
hmm.. Well, there goes that idea..

**Comment by Cloud9Developer** [comment]
see, here's your logs:

```
[NLP]: [2024-06-04] :: [12:25:47] - [INFO] Parsing Series..
[NLP]: [2024-06-04] :: [12:25:47] - [INFO] Scan Size: 12080
[NLP]: [2024-06-04] :: [12:25:47] - [INFO] Scanning 'Series'
[NLP]: [2024-06-04] :: [12:25:47] - [ERR] An error has occured: System.IndexOutOfRangeException: Index was outside the bounds of the array.
```

But there should be another line in there, like this (at the very least):
```
[NLP]: [2024-06-04] :: [12:25:47] - [INFO] Parsing Series..
[NLP]: [2024-06-04] :: [12:25:47] - [INFO] Scan Size: 12080
[NLP]: [2024-06-04] :: [12:25:47] - [INFO] Scanning 'Series'
[NLP]: [2024-06-04] :: [12:25:47] - ---------------
[NLP]: [2024-06-04] :: [12:25:47] - [ERR] An error has occured: System.IndexOutOfRangeException: Index was outside the bounds of the array.
```

Which tells me that it's crashing when doing this `foreach` in the code:

```
logger.Info($"Scanning '{type}'");
        foreach (BaseItem item in items)
        {
```

But i'm not sure why if you're file structure is following the recommendation...

**Comment by mastamjay** [comment]
could this have something to do with the german translation?

**Comment by Cloud9Developer** [comment]
Hmm, maybe? could be some special character that it's not liking

**Comment by mastamjay** [comment]
> Should I completely uninstall the newsletter plugin from jellyfin and also some files/folders of the plugin under Windows? If so, which folders and files of the plugin should I delete under Windows? Then I could test another version. Or is it enough to choose another version or plugin/catalogue and install it?

Should I try this out

**Comment by Cloud9Developer** [comment]
You can attempt it, but I doubt it would do anything except cause you headache having to reconfigure everything again

**Comment by Cloud9Developer** [comment]
Anyway you can swap to english translation to see if that's actually the issue? If it works, we know it's due to translation. If not, then more thought needs to be put into what's going on lol

**Comment by Cloud9Developer** [comment]
My pleasure! I may be slow to respond or unresponsive at times due to work (@nylimited knows, he got annoyed at me a while back ;) lol) but I want to make sure it's working for as many people that I can

**Comment by nylimited** [comment]
> My pleasure! I may be slow to respond or unresponsive at times due to work (@nylimited knows, he got annoyed at me a while back ;) lol) but I want to make sure it's working for as many people that I can

I heard that work is a four lettered word ....    🤣
(Of course being a retired IT director lets me make fun of work)

**Comment by mastamjay** [comment]
Update:
got it running again.

1. What I did:
2. Windows 11 + Jellyfin Server 10.9.3
3. Newsletter Plugin v 0.6.2.1
4. Newsletter.db deleted in advance
5. then clicked on Filesystem Scraper (took longer because the whole library was captured)
6. Email sent -> works

However, among other things, the ratings and running times are not displayed correctly.

![grafik](https://github.com/Cloud9Developer/Jellyfin-Newsletter-Plugin/assets/54937544/95e12d5a-5691-4aea-ae20-e46b47c43948)
![grafik](https://github.com/Cloud9Developer/Jellyfin-Newsletter-Plugin/assets/54937544/7cfc56ff-8437-4da8-9a82-2c8465245ed6)

I also have a question as to whether I have used the modern templates for body and entry correctly

Body HTML:
<html>
<body>
	<div>
		<table style="margin-left: auto; margin-right: auto; font-family:'Google Sans',Arial,sans-serif;">
			<tr> 
				<td width="100%" height="100%" style="vertical-align: top; background-color: #000000;"> 
					<table id="InsertHere" name="MainTable" style="margin-left: auto; margin-right: auto; border-spacing: 0 10px; padding: 0 10px;"> 
						<tr style="text-align: center;"> 
							<td colspan="2"> 
								<h1 id="Title" style="font-size:3.3em;margin:10px 0 0 0;">
                                    <a href="{ServerURL}" target="_blank" style="color:#E197BC;text-decoration: none;">
                                        <img src="https://cdn0.iconfinder.com/data/icons/google-material-design-3-0/48/ic_play_circle_filled_white_48px-512.png" style="height: 1.3em;margin-bottom:-12px;"> 
                                        Jellyfin Newsletter
                                    </a>
                                </h1>
								<h3 id="Date" style="color:#FFFFFF;margin:0px;">{Date}</h3>
							</td> 
						</tr>
                        <tr>
                            
                            {EntryData}
                            
                        </tr>
					</table> 
				</td>
			</tr> 
		</table> 
	</div>
</body>
</html>

Entrydata HTML:

 
							<tr class="boxed" style="outline: 1px solid #bf7df0;border-radius: 25px;">
								<td>
									<table style="margin:0;padding:0;border:0;">
										<tr style="margin:0;padding:0;border:0;">
											<td rowspan="0" class="lefttable" style="padding-right: 20px; padding-left: 20px; padding-top: 20px; padding-bottom: 20px;"> 
												<a href="{ServerURL}" target="_blank"><img style="width: 133px; height: 200px;" src="{ImageURL}"></a>
											</td>
											<td class="righttable" style="vertical-align: top; padding-right: 10px; padding-top: 10px;width:100%;"> 
												<div id="SeriesTitle" class="text" style="color: #FFFFFF; text-align: center;">
													<h2 style="margin-top:6px;margin-bottom:0px;"><a href="{ServerURL}" target="_blank" style="color:#87beff;text-decoration: none;">
													{Title} 
													</a></h2>
												</div>
												
												
												<table>
													<td>
														<div id="runtime" style="color: #d190c5; font-size: small;">
															<b>Duration:</b> <i>{RunTime} min.</i> |
														</div>
													</td>
													<td>
														<div id="communityrating" style="color: #d190c5; font-size: small;">
															<img src="https://cdn-icons-png.flaticon.com/512/3418/3418886.png" style="height: 15px;"> 
														</div>
													</td>
													<td>
														<div id="communityrating" style="color: #d190c5; font-size: small;">
															{CommunityRating} |
														</div>
													</td>
													<td>
														<div id="officialrating" style="color: #d190c5; font-size: small;">
															{OfficialRating} |
														</div>
													</td>
													<td>
														<div id="premiereyear" style="color: #d190c5; font-size: small;">
															{PremiereYear}
														</div>
													</td>
												</table>
							
												
												
												<div class="text" style="color: #d190c5;">
													{SeasonEpsInfo}
												</div>
												<hr style="border-color: #007fb1;margin-bottom:12px;width:80%;"/> 
												<div id="Description" class="text" style="color: #FFFFFF;">
													{SeriesOverview}
												</div>
											</td>
										</tr>
									</table>
								</td>
							</tr>

**Comment by p37307** [comment]
> > Not sure if this is what is going on with you, but I looked in the config file `jellyfin\plugins\configurations\Jellyfin.Plugin.Newsletters.xml` for Windows 10 and had to change `<NewsletterDir>` path from `\cache\temp/Newsletters/` to `\cache\temp\Newsletters\` and manually created the Newsletters folder.
> > It worked after that.
> 
> @p37307 Ahh, looks like unix paths and windows don't mesh.. Can you open an issue for this? I'll see if I can find a way to get that fixed

@Cloud9Developer Sure will. It'll be a little bit later this evening.

**Comment by mastamjay** [comment]
I have now tested all versions above 0.6.2.1 and have to realise that something has changed since 0.6.3.0 that is not compatible with Jellyfin Server for Windows.

**Comment by Cloud9Developer** [comment]
0.6.3.0 was the release that updated support for jellyfin 10.9.0

**Comment by Cloud9Developer** [comment]
What version of jellyfin are you on?

**Comment by mastamjay** [comment]
> What version of jellyfin are you on?

10.9.3

**Comment by Cloud9Developer** [comment]
gotcha. Yeah, I'm not sure what the problem is. Probably something to do with the versioning compatibility with windows server like you said, but I can't say for sure..

**Comment by mastamjay** [comment]
Thanks for the work on your newsletter plugin. Can't wait to see what else is coming. Can I buy you a coffee?

**Comment by nylimited** [comment]
Today's sample for you (attached) ..  Missed one TV show and associated picture. Okay, missed all data except the fact that *something* should have been there. 


[newsletter.zip](https://github.com/user-attachments/files/15844725/newsletter.zip)
 And, yes, separating movies and TV and music is becoming kinda important ...
